### PR TITLE
[mypyc] Add primitive ops for dictionary methods

### PR DESCRIPTION
**Pull request by ilevkivskyi**
This covers both view and list versions (it looks like both are used relatively often, at least in mypy). This accompanies https://github.com/python/mypy/pull/8725 to account for cases where `keys`/`values`/`items` appear in non-loop contexts.

### CODE REVIEW

**Comment by ilevkivskyi** [comment]
...and as usual Python 3.5 is totally busted. I will probably just disable this on Python 3.5. PR #8725 adds the corresponding `CompilerOptions` flag, I will just reuse it here.

**Comment by ilevkivskyi** [comment]
@msullivan @JukkaL I switched to disabling the optimization for subclasses here too. Now tests pass on Python 3.5.

**Comment by JukkaL** [comment]
I measured that this can make the ops at least 2x faster for small dictionaries. This is a great improvement!

File: mypyc/test-data/run.test [hunk 1]
```diff
@@ -1035,14 +1035,17 @@ def u(x: int) -> int:
     d.update(x=x)
     return d['x']
 
+def get_content(d: Dict[int, int]) -> Tuple[List[int], List[int], List[Tuple[int, int]]]:
+    return list(d.keys()), list(d.values()), list(d.items())
```
Comment (JukkaL) [inline mypyc/test-data/run.test:1039]: Add also tests for the "plain" primitive ops that don't create a list, such as by doing `set(d.keys())`?

Might be worth testing dict subclasses as well.

**Pull request merged**